lib/theme_check/language_server/variable_lookup_finder.rb

```diff
@@ -0,0 +1,314 @@
+        is_in_variable_segment = previous_char =~ VARIABLE_LOOKUP_CHARACTERS
+        is_on_blank_variable_lookup_position = content[0..cursor - 1] =~ /[{:,-]\s+$/
+        (
+          is_liquid_variable && (
+            is_in_variable_segment ||
+            is_on_blank_variable_lookup_position
+          )
+        )
+      end
+
+      def lookup_liquid_variable(content, cursor)
+        return unless cursor_is_on_liquid_variable_lookup_position(content, cursor)
+        start_index = content.match(/#{Liquid::VariableStart}-?/o).end(0) + 1
+        end_index = cursor - 1
+
+        # We take the following content
+        # - start after the first two {{
+        # - end at cursor position
+        #
+        # That way, we'll have a partial liquid variable that
+        # can be parsed such that the "last" variable_lookup
+        # will be the one we're trying to complete.
+        markup = content[start_index..end_index]
+
+        # Early return for incomplete variables
+        return empty_lookup if markup =~ /\s+$/
+
+        # Now we go to hack city... The cursor might be in the middle
+        # of a string/square bracket lookup. We need to close those
+        # otherwise the variable parse won't work.
+        markup += "'" if markup.count("'").odd?
+        markup += '"' if markup.count('"').odd?
+        markup += "]" if markup =~ UNCLOSED_SQUARE_BRACKET
+
+        variable = variable_from_markup(markup)
+
+        variable_lookup_for_liquid_variable(variable)
+      end
+
+      def cursor_is_on_liquid_tag_lookup_position(content, cursor)
+        markup = content[0..cursor - 1]
+        is_liquid_tag = content.match?(Liquid::TagStart)
+        is_in_variable_segment = markup =~ ENDS_WITH_POTENTIAL_LOOKUP
+        is_on_blank_variable_lookup_position = markup =~ ENDS_WITH_BLANK_POTENTIAL_LOOKUP
+        (
+          is_liquid_tag && (
+            is_in_variable_segment ||
+            is_on_blank_variable_lookup_position
+          )
+        )
+      end
+
+      # Context:
+      #
+      # We know full well that the code as it is being typed is probably not
+      # something that can be parsed by liquid.
+      #
+      # How this works:
+      #
+      # 1. Attempt to turn the code of the token until the cursor position into
+      #    valid liquid code with some hacks.
+      # 2. If the code ends in space at a "potential lookup" spot
+      #   a. Then return an empty variable lookup
+      # 3. Parse the valid liquid code
+      # 4. Attempt to extract a VariableLookup from Liquid::Template
+      def lookup_liquid_tag(content, cursor)
+        return unless cursor_is_on_liquid_tag_lookup_position(content, cursor)
+
+        markup = parseable_markup(content, cursor)
+        return empty_lookup if markup == :empty_lookup_markup
+
+        template = Liquid::Template.parse(markup)
+        current_tag = template.root.nodelist[0]
+
+        case current_tag.tag_name
+        when "if", "unless"
```

Comment:
Would `when` work here too?

Reply:
When is gets parsed as `case`. (I grab the when from the contents)